### PR TITLE
Feature/add javascript linting

### PR DESCRIPTION
**Pull request by frontendjim**
This includes a git pre-commit hook to prevent a commit of code if linting errors exist

### CODE REVIEW

File: lib/modules/vacancy/index.js [hunk 1]
```diff
@@ -21,84 +21,99 @@ const LOCATION_RADIUS_OPTIONS = [
...
-
-    const url = `${ process.env.API_URL }:${process.env.API_PORT}/vacancy/search?size=${size}&page=${page-1}`;
+const fetchVacancyList = async function ({
+    location = '',
+    keyword = '',
+    depts = [],
+    size = '10',
+    radius = DEFAULT_LOCATION_RADIUS, page = 0,
+}) {
+    const url = `${process.env.API_URL}:${process.env.API_PORT}/vacancy/search?size=${size}&page=${page - 1}`;
```
Comment (reachsaurabh) [inline lib/modules/vacancy/index.js]: externalize this when we config server and eureka running

File: lib/modules/vacancy/index.js [hunk 2]
```diff
@@ -21,84 +21,99 @@ const LOCATION_RADIUS_OPTIONS = [
...
+                radius,
+            },
+            department: typeof depts === 'string' ? [depts] : depts,
+        }),
     };
 
     try {
         return await fetch(url, options).then(response => response.json());
-    } catch(e) {
+    } catch (e) {
```
[the remark this answers is not in the document]
Comment (frontendjim) [reply lib/modules/vacancy/index.js]: It didn't flag it. This issues needs looking as part of a larger error handling piece.

File: lib/modules/vacancy/index.js (hunk 1 above)
Comment (frontendjim) [reply lib/modules/vacancy/index.js]: Yep.

File: public/javascripts/main.js [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
+        mediaQuery: 'only screen and (max-width: 640px)',
+        iconPosition: 'after',
+    });
 }
 
+// eslint-disable-next-line no-undef
 const showHideFields = document.getElementsByClassName('js-showhide-field');
-for(var i = 0; i < showHideFields.length; i++) {
-    new ShowHide({
+for (let i = 0; i < showHideFields.length; i += 1) {
```
Comment (reachsaurabh) [inline public/javascripts/main.js]: is there no better for each contruct , this looks too primitive .

Comment (frontendjim) [reply public/javascripts/main.js]: Changed and pushed a new commit

File: lib/modules/vacancy/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (gmackay) [reply lib/modules/vacancy/index.js]: The space after the catch help with readability.

File: lib/modules/date/date.test.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-test('date is formatted corrently', t => {
+test('date is formatted corrently', (t) => {
     const beforeDate = '2018-05-10 00:00:00';
     const afterDate = '10 May 2018';
 
     t.deepEqual(gdsDateFormat(beforeDate), afterDate);
 });
 
-test('date is formatting corrently and day is single digit', t => {
+test('date is formatting corrently and day is single digit', (t) => {
```
Comment (gmackay) [inline lib/modules/date/date.test.js]: Love these brackets

**Pull request merged**